frontend/views/containers/settings/UserProfile.vue

```diff
@@ -2,81 +2,62 @@
   .settings-container
     p.username @{{ ourUsername }}
 
-    section.card
-      form(
-        ref='ProfileForm'
-        name='ProfileForm'
-        @submit.prevent='save'
-      )
-        p.is-success(
-          v-if='ephemeral.profileSaved'
-          data-test='profileSaveSuccess'
-        )
-          i.icon-check
-          i18n Profile saved successfully!
-
-        p.error(v-if='ephemeral.errorMsg') {{ ephemeral.errorMsg }}
-
-        .avatar(:class='{ error: $v.form.picture.$error }')
-          label(for='profilePicture')
-            avatar(
-              :src='userPictureInitial'
-              ref='picture'
-            )
-            i18n.link Change avatar
+    form(@submit.prevent='')
+      .c-avatar
+        label.c-avatar-field
+          avatar.c-avatar-img(
+            :src='attributes.picture'
+            ref='picture'
+          )
+          i18n.link.c-avatar-text Change avatar
 
-          input.profilePictureInput#profilePicture(
+          input.sr-only(
             type='file'
             name='profilePicture'
             accept='image/*'
             @change='fileChange($event.target.files)'
             placeholder='http://'
             data-test='profilePicture'
           )
+        // TODO #658
+        banner-scoped(ref='pictureMsg' data-test='avatarMsg')
 
-          i18n.error(tag='p' v-if='$v.form.picture.$error')
-            | The profile picture must be a valid url
-
-        .field
-          i18n.label(tag='label') Display Name
-
+    section.card
+      form(@submit.prevent='saveProfile')
+        label.field
+          i18n.label Display Name
           input.input(
             name='displayName'
             type='text'
             v-model='form.displayName'
             placeholder='Name'
             data-test='displayName'
           )
+          i18n.helper This is how others will see your name accross the platform.
 
-          i18n.help(tag='p') This is how others will see your name accross the platform.
-
-        .field
-          i18n.label(tag='label') Bio
-
+        label.field
+          i18n.label Bio
           textarea.textarea(
-            type='text'
             name='bio'
             v-model='form.bio'
             placeholder='Bio'
             data-test='bio'
           )
 
-        .field
-          i18n.label(tag='label') Email
-
+        label.field
+          i18n.label Email
```

Comment:
So, just noticing that the markup here has changed significantly:

- Line 47: a `<div>` has turned into a `<label>`
- Line 48: a `<label>` has turned into a `<span>`

And same elsewhere. You're certain this was intentional?

Reply:
Yes, this is the most correct and simple way of coding form fields with a11y in mind. Every field should be surrounded by a `label` tag. Inside should have the label itself (ex: Email), the `input` and any other helpful text related to the field, ex: a helper or error message.
By having all this content inside the label, the SC will know that it's all related to the same thing, without the need to use `for` with unique `id`.